buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/TestProgressLogger.groovy

```diff
@@ -49,21 +53,81 @@ class TestProgressLogger implements AggregatedEventListener {
         progressLogger = factory.newOperation(TestProgressLogger)
         progressLogger.setDescription('Randomized test runner')
         progressLogger.started()
-        progressLogger.progress('Starting JUnit4 with ' + e.getSlaveCount() + ' jvms')
+        progressLogger.progress("Starting JUnit4 with $e.slaveCount jvms")
+    }
+
+    @Subscribe
+    void onTestResult(AggregatedTestResultEvent e) throws IOException {
+        testsCompleted.incrementAndGet()
+        switch (e.status) {
+        case ERROR:
+        case FAILURE:
+            testsFailed.incrementAndGet()
+            break
+        case IGNORED:
+        case IGNORED_ASSUMPTION:
+            testsIgnored.incrementAndGet()
+            break
+        case OK:
+            break
+        default:
+            throw new IllegalArgumentException(
+                "Unknown test status: [$e.status]")
+        }
+        String suiteName = e.description.displayName
+        log(new SuiteInfo(
+            name: suiteName.substring(suiteName.lastIndexOf('.') + 1,
+                    suiteName.length() - 1),
+            slave: e.slave.id,
+        ))
     }
 
     @Subscribe
     void onSuiteResult(AggregatedSuiteResultEvent e) throws IOException {
-        final int suitesCompleted = suitesCompleted.incrementAndGet();
-        final int testsCompleted = testsCompleted.addAndGet(e.getDescription().testCount())
-        final int testsFailed = testsFailed.addAndGet(e.getErrorCount() + e.getFailureCount())
-        final int testsIgnored = testsIgnored.addAndGet(e.getIgnoredCount())
-        Description description = e.getDescription()
-        String suiteName = description.getDisplayName();
-        suiteName = suiteName.substring(suiteName.lastIndexOf('.') + 1);
-        progressLogger.progress('Suites [' + suitesCompleted + '/' + totalSuites + '], Tests [' + testsCompleted + '|' + testsFailed + '|' + testsIgnored + '], ' + suiteName + ' on J' + e.getSlave().id + ' in ' + formatDurationInSeconds(e.getExecutionTime()))
+        suitesCompleted.incrementAndGet();
+        String suiteName = e.description.displayName
+        SuiteInfo suiteInfo = new SuiteInfo(
+            name: suiteName.substring(suiteName.lastIndexOf('.') + 1),
+            slave: e.slave.id,
+            duration: formatDurationInSeconds(e.executionTime),
+        )
+        lastSuite.set(suiteInfo)
+        log(suiteInfo)
+    }
+
+    private void log(SuiteInfo currentSuite) {
+        SuiteInfo lastSuite = this.lastSuite.get()
+        /* Only use the suite from the current event if we haven't completed a
+          suite yet. Using the current suite at all makes the results useful
+          for long suites like RestIT and using the last completed suite most
+          of the time keeps the suite from jumping back and forth. */
+        String suiteInfo
+        if (lastSuite == null) {
+            suiteInfo = ", current suite is $currentSuite"
```

Comment:
When a test completes, do we get the rest test name? Maybe we could instead make an option to this logger (that can be set in testLogging gradle test logging config) on whether the test name should be output on completion? So then we could enable that in RestIntegTestTask.